pandas/tests/frame/test_mutate_columns.py

```diff
@@ -89,11 +89,35 @@ def test_assign_bad(self):
             df.assign(lambda x: x.A)
         with pytest.raises(AttributeError):
             df.assign(C=df.A, D=df.A + df.C)
+
+    @pytest.mark.skipif(PY36, reason="""Issue #14207: valid for python
+                        3.6 and above""")
+    def test_assign_bad_old_version(self):
```

Comment:
can you name this something more relevant

Reply:
do you mean the name of the function or the reason?

Reply:
name of the function